### PR TITLE
feat(docs): Document the iframe protocol.

### LINKED ISSUE #2026: views should rely on the Relier model to fetch relier data, not the URL search params
Now that we have the ever so awesome [Relier models](https://github.com/mozilla/fxa-content-server/tree/master/app/scripts/models/reliers), we should fetch data from those instead of from the query parameters. Fetching data from both places could lead to [cache coherency problems](http://cmsc411.com/multiprocessors/cache-coherence-problem) if the relier is updated but the query parameter is not.

> ➜  views git:(master) rgrep "searchParam" .
./base.js:      this.automatedBrowser = !!this.searchParam('automatedBrowser');
./base.js:    searchParam: function (itemName) {
./base.js:      return Url.searchParam(itemName, this.window.location.search);
./base.js:      this[itemName] = this.searchParam(itemName);
./reset_password.js:      return this.searchParam('email') || Session.prefillEmail || '';
./settings.js:      var uid = self.searchParam('uid');
./sign_in.js:      this.prefillEmail = Session.prefillEmail || this.searchParam('email');
./sign_up.js:      var email = Session.prefillEmail || this.searchParam('email');
➜  views git:(master)

Email verification data is imported as well, though this does not make sense to put on the relier:
> ➜  views git:(master) ✗ rgrep "importSearch" .
./base.js:    importSearchParam: function (itemName) {
./complete_reset_password.js:        this.importSearchParam('token');
./complete_reset_password.js:        this.importSearchParam('code');
./complete_reset_password.js:        this.importSearchParam('email');
./complete_sign_up.js:        this.importSearchParam('uid');
./complete_sign_up.js:        this.importSearchParam('code');

### PR DESCRIPTION
**Pull request by shane-tomlinson**
fixes #2026

### CODE REVIEW

File: docs/relier-communication-protocols/iframe.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+* `redirect` {String} - The URI that the user would have been redirected to had the redirect flow been used.
+
+#### oauth_cancel
+Sent when a user indicates they want to close the iframe. The parent should close the iframe.
+
+### Command format
+Since all commands are sent over postMessage, commands and their data must be represented by strings. The format is:
+
+```
+command!!!data
```
Comment (zaach) [inline docs/relier-communication-protocols/iframe.md]: @ckarlof was curious why this isn't JSON, i.e. `JSON.stringify({ command: "command", data: "{ ... }" }))`

**Comment by zaach** [comment]
:+1:

File: docs/relier-communication-protocols/iframe.md (hunk 1 above)
Comment (shane-tomlinson) [inline docs/relier-communication-protocols/iframe.md]: > @ckarlof was curious why this isn't JSON, i.e. JSON.stringify({ command: "command", data: "{ ... }" }))

I know, he asked me the same yesterday. The honest answer is I didn't think of it.

Comment (ckarlof) [inline docs/relier-communication-protocols/iframe.md]: No worries. We'll laugh about it over a beer sometime!

**Pull request merged**